internal/services/oracle/oracle_database_system_versions_data_source.go

```diff
@@ -0,0 +1,170 @@
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2025-09-01/dbversions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type DatabaseVersionsDataSource struct{}
+
+type DatabaseVersionsModel struct {
+	Location                            string                     `tfschema:"location"`
+	DatabaseSystemShape                 string                     `tfschema:"database_system_shape"`
+	DatabaseSoftwareImageSupportEnabled *bool                      `tfschema:"database_software_image_support_enabled"`
```

Comment:
a pointer of a primitive type should not be used in the model
```suggestion
	DatabaseSoftwareImageSupportEnabled bool                      `tfschema:"database_software_image_support_enabled"`
```